firmware-support/bittide-sys/src/ugn.rs

```diff
@@ -0,0 +1,12 @@
+// SPDX-FileCopyrightText: 2024 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use ufmt::derive::uDebug;
+
+#[repr(C)]
+#[derive(uDebug, PartialEq, Eq, Copy, Clone)]
+pub struct Ugn {
+    pub tx_counter: u64,
+    pub rx_counter: u64,
+}
```

Comment:
Can you add some utility functions that:
* Detect if we have captured the UGNs (Both counters are 0 => Not captured). This could also be used to see if the link is up.
* Subtract the UGNs correctly to calculate the IGN.


Furthermore, change names to `local` and `remote` to be consistent with the Clash code